src/psyclone/psyGen.py

```diff
@@ -6117,26 +6117,88 @@ def reference_accesses(self, var_accesses):
 
 class Literal(Node):
     '''
-    Node representing a Literal
+    Node representing a Literal.
 
-    :param str value: String representing the literal value.
+    :param value: the value of the literal.
+    :type value: str (for numberical values) or bool
+    :param datatype: The datatype of this literal.
+    :type datatype: :py:class:`psyclone.psyGen.DataType`
     :param parent: the parent node of this Literal in the PSyIR.
     :type parent: :py:class:`psyclone.psyGen.Node`
     '''
-    def __init__(self, value, parent=None):
+    # A Literal cannot have DEFERRED type
+    VALID_DATA_TYPES = [DataType.INTEGER, DataType.REAL,
+                        DataType.CHARACTER, DataType.BOOLEAN]
+
+    def __init__(self, value, datatype, parent=None):
         super(Literal, self).__init__(parent=parent)
-        self._value = value
+        self.datatype = datatype
+        self.value = value
 
     @property
-    def value(self):
+    def datatype(self):
+        '''
+        :returns: the type of this Literal.
+        :rtype: :py:class:`psyclone.psyGen.DataType`
+        '''
+        return self._datatype
+
+    @datatype.setter
```

Comment:
Is this setter used anywhere, apart from the __init__ of this class? Similary for the value setter. If not, perhaps it makes sense for Literals to be immutable?

Reply:
Yes, good idea.